packages/windows/src/RPA/Windows/keywords/window.py

```diff
@@ -367,14 +376,64 @@ def close_current_window(self) -> bool:
 
             ${status}=  Close Current Window
         """
-        if not self.ctx.window:
-            self.ctx.logger.warning("There is no active window")
+        window = self.window
+        if window is None:
+            self.logger.warning("There is no active window!")
+            self.ctx.window_element = None
             return False
-        pid = self.ctx.window.item.ProcessId
-        name = self.ctx.window.item.Name
-        self.ctx.logger.info(
-            'Closing window with Name:"%s", ProcessId: %s' % (name, pid)
-        )
+
+        pid = window.item.ProcessId
+        self.logger.info("Closing window with name: %s (PID: %d)", window.name, pid)
         os.kill(pid, signal.SIGTERM)
-        self.ctx.window = None
+        self.ctx.window_element = None
+
+        anchor = self.ctx.anchor_element
+        if anchor and window.is_sibling(anchor):
+            # We just closed the anchor (along with its relatives), so clear it out
+            #  properly.
+            self.ctx.clear_anchor()
+
         return True
+
+    @keyword(tags=["window"])
+    @with_timeout
+    def close_window(
+        self,
+        locator: Optional[Locator] = None,
+        timeout: Optional[float] = None,  # pylint: disable=unused-argument
+    ) -> int:
+        """Closes identified windows or logs the problems.
+
+        :param locator: String locator or `Control` element.
+        :return: How many windows were found and closed.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            ${closed_count} =     Close Window    Calculator
+        """
+        # Starts the search from Desktop level.
+        root_element = WindowsElement(auto.GetRootControl(), locator)
+        # With all flavors of locators. (if flexible)
+        elements: Optional[List[WindowsElement]] = None
+        for loc in self._iter_locator(locator):
+            try:
+                elements = self.ctx.get_elements(loc, root_element=root_element)
+            except (ElementNotFound, LookupError):
+                pass
+            else:
+                break
+        if not elements:
+            self.logger.info("Couldn't find any window with locator: %s", locator)
+            return 0
+
+        closed = 0
+        for element in elements:
+            self.logger.debug("Controlling and closing window: %s", element)
+            try:
+                self.control_window(element)
```

Comment:
what's the "controlled window" after these close calls? sorry if that's obvious from reading the library docs

Reply:
- Each time a window is controlled -> it becomes the active window (only one active/controlled window can exist at a time)
- Each active window gets closed -> the current `self.ctx.window_element` becomes None (after closing per se)

So no matter how many close calls (>=1) you have, in the end there's no current active window set. (as `Close Window` is more like a wrapper over the following: find me all the windows matching `locator`, then activate and close each of them, one by one)